File: acmp.ru/[0001] - [0100]/[0053] Раскраска таблицы умножения.py
n, m = map(int, input().split())

red = green = blue = black = 0


for i in range(1, n + 1):
    for j in range(1, m + 1):
        a = i * j

        if a % 5 == 0:
            blue += 1
        elif a % 3 == 0:
            green += 1
        elif a % 2 == 0:
            red += 1
        else:
            black += 1

print('RED :', red)
print('GREEN :', green)
print('BLUE :', blue)
print('BLACK :', black)

File: acmp.ru/[0001] - [0100]/[0053] Раскраска таблицы умножения.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m, red = 0, green = 0, blue = 0, black = 0, t;
    cin >> n >> m;

    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= m; j++) {
            t = i * j;

            if (t % 5 == 0)
                blue++;
            else if (t % 3 == 0)
                green++;
            else if (t % 2 == 0)
                red++;
            else
                black++;
        }
    }

    printf("RED : %d\nGREEN : %d\nBLUE : %d\nBLACK : %d\n",
           red, green, blue, black);

    return 0;
}
